.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+name: PR Check CI
+
+on:
+  pull_request:
```

Comment:
Should we not also run on the  `push` event?  Also I think this restricts to pull requests off master which might not be what we want if there is a long running feature branch.  I think is can just be this.
`on: [push, pull_request]`

Reply:
I see that it by default includes synchronize so probably fine to leave it, but I would drop the branch.  Maybe we still want to keep the push but limited to branch master so there is always a build on master 
something like this:
```
on:
  pull_request
  push:
    branches: [master]
```

Reply:
> @liuguo09 also if you have your Dockerfile somewhere I would be happy to submit a PR against it to improve it a bit. There is some low hanging fruit that would reduce the size by quite a bit.

Yes, that's great. I meant to add the Dockerfile into nuttx testing repo. I'll add it later. Then you could update it there.

Reply:
> I see that it by default includes synchronize so probably fine to leave it, but I would drop the branch. Maybe we still want to keep the push but limited to branch master so there is always a build on master
> something like this:
> 
> ```
> on:
>   pull_request
>   push:
>     branches: [master]
> ```

Yes, besides master branch, pull request check for feature branch would also be useful. 
Now I use the pull request git log "Merge new-commit-id into old-commit-id" to filter commits in the PR and then do checkpatch.sh. But for master push, it should not work and skip. Just see the github action _job steps.if_ could be used to judge it. I'll update it later.

Reply:
Update on trigger pull request  for all branches. As to push event, I think it may be not necessary since there is no committer push master directly.

Reply:
I think it captures the merge situation which is what I'm worried about.

Reply:
There is also the nightly build to make the full build although it's per day.